src/index.js

```diff
@@ -1,65 +1,172 @@
+
 /**
- * App ID for the skill
+ * This sample demonstrates a simple skill built with the Amazon Alexa Skills Kit.
+ * The Intent Schema, Custom Slots, and Sample Utterances for this skill, as well as
+ * testing instructions are located at http://amzn.to/1LzFrj6
+ *
+ * For additional samples, visit the Alexa Skills Kit Getting Started guide at
+ * http://amzn.to/1LGWsLG
  */
 
-var APP_ID = "amzn1.echo-sdk-ams.app.23ed8a00-90ef-4ba5-b418-a1d4d8ec5e32";
-
-var AlexaSkill = require('./AlexaSkill');
-
-
+// Route the incoming request based on type (LaunchRequest, IntentRequest,
+// etc.) The JSON body of the request is provided in the event parameter.
 
-var ReverseGeek = function () {
-    AlexaSkill.call(this, APP_ID);
+exports.handler = function (event, context) {
+    try {
+        console.log("event.session.application.applicationId=" + event.session.application.applicationId);
+
+
+        if (event.session.application.applicationId !== "amzn1.echo-sdk-ams.app.23ed8a00-90ef-4ba5-b418-a1d4d8ec5e32") {
+             context.fail("Invalid Application ID");
+        }
+
+        if (event.session.new) {
+            onSessionStarted({requestId: event.request.requestId}, event.session);
+        }
+
+        if (event.request.type === "LaunchRequest") {
+            onLaunch(event.request,
+                event.session,
+                function callback(sessionAttributes, speechletResponse) {
+                    context.succeed(buildResponse(sessionAttributes, speechletResponse));
+                });
+        } else if (event.request.type === "IntentRequest") {
+            onIntent(event.request,
+                event.session,
+                function callback(sessionAttributes, speechletResponse) {
+                    context.succeed(buildResponse(sessionAttributes, speechletResponse));
+                });
+        } else if (event.request.type === "SessionEndedRequest") {
+            onSessionEnded(event.request, event.session);
+            context.succeed();
+        }
+    } catch (e) {
+        context.fail("Exception: " + e);
+    }
 };
 
-// Extend AlexaSkill
-ReverseGeek.prototype = Object.create(AlexaSkill.prototype);
-ReverseGeek.prototype.constructor = ReverseGeek;
-
-ReverseGeek.prototype.eventHandlers.onLaunch = function (launchRequest, session, response) {
+function onSessionStarted(sessionStartedRequest, session) {
+    console.log("onSessionStarted requestId=" + sessionStartedRequest.requestId +
+        ", sessionId=" + session.sessionId);
+}
+
+function onLaunch(launchRequest, session, callback) {
+    console.log("onLaunch requestId=" + launchRequest.requestId +
+        ", sessionId=" + session.sessionId);
+
+     getWelcomeResponse(callback);
```

Comment:
Very close.

Your lambda function [checks the request type](https://github.com/javierforero/reverse_geek/blob/d9d8d7dbf411b8fbc9a081a7ed82fc05b47c2fb6/src/index.js#L27) and calls `getWelcomeResponse` if its a launch intent.

You define and pass it a callback function on [line 30](https://github.com/javierforero/reverse_geek/blob/d9d8d7dbf411b8fbc9a081a7ed82fc05b47c2fb6/src/index.js#L30-L32)

Yes, the rest is correct.  You call `context.succeed` to indicate that the Lambda can complete and pass it a response.